src/super_gradients/training/metrics/pose_estimation_metrics.py

```diff
@@ -0,0 +1,288 @@
+        if len(groundtruths) != len(gt_areas) != len(gt_bboxes) != len(gt_is_crowd):
+            raise ValueError(
+                "Length of groundtruths, areas, bboxes and iscrowd should be equal. Got {} and {} and {} and {}".format(
+                    len(groundtruths), len(gt_areas), len(gt_bboxes), len(gt_is_crowd)
+                )
+            )
+
+        predicted_poses = torch.tensor(predicted_poses, dtype=torch.float, device=self.device)
+        predicted_scores = torch.tensor(predicted_scores, dtype=torch.float, device=self.device)
+
+        gt_keypoints = torch.tensor(groundtruths, dtype=torch.float, device=self.device)
+        gt_areas = torch.tensor(gt_areas, dtype=torch.float, device=self.device)
+        gt_bboxes = torch.tensor(gt_bboxes, dtype=torch.float, device=self.device)
+
+        gt_keypoints_xy = gt_keypoints[:, :, 0:2]
+        gt_keypoints_visibility = gt_keypoints[:, :, 2]
+        gt_all_kpts_invisible = gt_keypoints_visibility.eq(0).all(dim=1)
+        gt_is_crowd = torch.tensor(gt_is_crowd, dtype=torch.bool, device=self.device)
+        gt_is_ignore = gt_all_kpts_invisible | gt_is_crowd
+
+        try:
+            targets = gt_keypoints_xy[~gt_is_ignore] if len(groundtruths) else []
+            targets_visibilities = gt_keypoints_visibility[~gt_is_ignore] if len(groundtruths) else []
+            targets_areas = gt_areas[~gt_is_ignore] if len(groundtruths) else []
+            targets_bboxes = gt_bboxes[~gt_is_ignore]
+            targets_ignored = gt_is_ignore[~gt_is_ignore]
+
+            crowd_targets = gt_keypoints_xy[gt_is_ignore] if len(groundtruths) else []
+            crowd_visibilities = gt_keypoints_visibility[gt_is_ignore] if len(groundtruths) else []
+            crowd_targets_areas = gt_areas[gt_is_ignore]
+            crowd_targets_bboxes = gt_bboxes[gt_is_ignore]
+            crowd_targets_ignored = gt_is_ignore[gt_is_ignore]
+
+            preds_matched, preds_to_ignore, preds_scores, num_targets = compute_img_keypoint_matching(
+                predicted_poses,
+                predicted_scores,
+                #
+                targets=targets,
+                targets_visibilities=targets_visibilities,
+                targets_areas=targets_areas,
+                targets_bboxes=targets_bboxes,
+                targets_ignored=targets_ignored,
+                #
+                crowd_targets=crowd_targets,
+                crowd_visibilities=crowd_visibilities,
+                crowd_targets_areas=crowd_targets_areas,
+                crowd_targets_bboxes=crowd_targets_bboxes,
+                crowd_targets_ignored=crowd_targets_ignored,
+                #
+                iou_thresholds=self.iou_thresholds,
+                sigmas=self.oks_sigmas,
+                top_k=self.max_objects_per_image,
+            )
+        except IndexError as e:
+            raise e
```

Comment:
Is there an advantage of catching it if you raise it with no change right after?

Reply:
No, that's a leftover. Thanks for spotting this)